scripts/tests/harness.sh

```diff
@@ -47,6 +47,21 @@ function forest_run_node_detached {
   $FOREST_PATH --chain calibnet --encrypt-keystore false --log-dir "$LOG_DIRECTORY" --detach --save-token ./admin_token --track-peak-rss
 }
 
+function forest_run_node_stateless_detached {
+  CONFIG_PATH="./stateless_forest_config.toml"
+  echo "${CONFIG_PATH}"
+  echo "Running forest in stateless and detached mode"
+  {
+    echo "[client]"
+    echo "data_dir = \"/tmp/stateless_forest_data\""
+    echo ""
+    echo "[network]"
+    echo "listening_multiaddrs = [\"/ip4/127.0.0.1/tcp/0\"]"
+  } > $CONFIG_PATH
+  
+  $FOREST_PATH --detach --chain calibnet --encrypt-keystore false --config "$CONFIG_PATH" --log-dir "$LOG_DIRECTORY" --save-token ./stateless_admin_token --no-gc --stateless
```

Comment:
You used `--no-gc` and `--stateless`. Should this be implied in the code?

Reply:
I'm not sure if we should run `gc` by default to keep the disk usage of a stateless node bounded.  @lemmih what's your suggestion?

Reply:
Conceptually, a stateless node should not use the database at all. A stateless node must not use an unbounded amount of disk space. Only N blocks should be cached (ideally in memory); all other blocks should be forgotten.

Reply:
> Conceptually, a stateless node should not use the database at all. A stateless node must not use an unbounded amount of disk space. Only N blocks should be cached (ideally in memory); all other blocks should be forgotten.

@lemmih Should in-memory store change be included in this PR? I feel it might have quite some conflicts with the ongoing GC refactoring PR, or should I do that after GC PR is merged?

Reply:
Sure, you can solve #3556 with multiple PRs.